### PR TITLE
fix(Combobox): remove "defaultValue" from hooks dependencies

### PR DESCRIPTION
**Pull request by layershifter**
## Previous Behavior

#31513 pointed out that `props.defaultValue` is used as a dependency in `useMemo()` and on its change - will break a component in some cases.

## New Behavior

`props.defaultValue` is not a dependency anymore and its update won't cause issues.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `104.398 kB`<br />`34.199 kB` | `104.383 kB`<br />`34.197 kB` | `-15 B` <br />`-2 B` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `104.984 kB`<br />`34.114 kB` | `104.969 kB`<br />`34.111 kB` | `-15 B` <br />`-3 B` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.097 MB`<br />`270.957 kB` | `1.097 MB`<br />`270.954 kB` | `-15 B` <br />`-3 B` |
| <samp>react-tag-picker</samp> <br /> <abbr title='bundle-size/Package.fixture.js'>@fluentui/react-tag-picker - package</abbr> | `182.176 kB`<br />`54.779 kB` | `182.161 kB`<br />`54.777 kB` | `-15 B` <br />`-2 B` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `107.387 kB`<br />`35.758 kB` | `107.372 kB`<br />`35.755 kB` | `-15 B` <br />`-3 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.14 kB`<br />`20.137 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `218 kB`<br />`63.151 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.447 kB`<br />`14.59 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/b5e427c24efd04fbd54dd4e43819ddc24be0ac16'>b5e427c24efd04fbd54dd4e43819ddc24be0ac16</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/Avatar-mount_base.html">636</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/Avatar-mount.html">618</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/Button-mount_base.html">305</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/Button-mount.html">291</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/Field-mount_base.html">1165</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/Field-mount.html">1131</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/FluentProvider-mount_base.html">709</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/FluentProvider-mount.html">714</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/FluentProviderWithTheme-mount_base.html">84</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/FluentProviderWithTheme-mount.html">86</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">31</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">36</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">73</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/MakeStyles-mount_base.html">893</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/MakeStyles-mount.html">866</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/Persona-mount_base.html">1754</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/Persona-mount.html">1729</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/SpinButton-mount_base.html">1410</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/SpinButton-mount.html">1400</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/SwatchPicker-mount_base.html">1656</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32287/perf-test-react-components/SwatchPicker-mount.html">1618</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-combobox-1d78745f-864b-42ce-b821-16fbc1be84dc.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-combobox-1d78745f-864b-42ce-b821-16fbc1be84dc.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32287&commitId=8fe45cbba6ed94585e680bd5a1a8a60647f84060&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|7| Changed|
|Avatar Converged.badgeMask.chromium.png|11| Changed|

</details>


  <details><summary><b>Drawer</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.overlay drawer full.chromium.png|1139| Changed|
|Drawer.Full Overlay RTL.chromium.png|2890| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-combobox/library/src/utils/useComboboxBaseState.ts [hunk 2]
```diff
@@ -126,7 +126,7 @@ export const useComboboxBaseState = (
     // we do not want to accidentally override defaultValue on a second render
     // unless another value is intentionally set
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [controllableValue, editable, getOptionsMatchingValue, multiselect, props.defaultValue, selectedOptions]);
+  }, [controllableValue, editable, getOptionsMatchingValue, multiselect, selectedOptions]);
```
Comment (ling1726) [inline packages/react-components/react-combobox/library/src/utils/useComboboxBaseState.ts:129]: would be nice to have a test that changes defaultValue to make sure this doesn' t happen again

**Review by ling1726 — APPROVED** [review]

**Pull request merged**